examples/07_advanced/plot_beta_series.py

```diff
@@ -0,0 +1,389 @@
+    row_number : int
+        Row number in the DataFrame.
+        This indexes the trial that will be isolated.
+
+    Returns
+    -------
+    df : pandas.DataFrame
+        Update events information, with the select trial's trial type isolated.
+    trial_name : str
+        Name of the isolated trial's trial type.
+    """
+    df = df.copy()
+
+    # Determine which number trial it is *within the condition*
+    trial_condition = df.loc[row_number, 'trial_type']
+    trial_type_series = df['trial_type']
+    trial_type_series = trial_type_series.loc[
+        trial_type_series == trial_condition
+    ]
+    trial_type_list = trial_type_series.index.tolist()
+    trial_number = trial_type_list.index(row_number)
+
+    # We use a unique delimiter here (``__``) that shouldn't be in the
+    # original condition names.
+    # Technically, all you need is for the requested trial to have a unique
+    # 'trial_type' *within* the dataframe, rather than across models.
+    # However, we may want to have meaningful 'trial_type's (e.g., 'Left_001')
+    # across models, so that you could track individual trials across models.
+    trial_name = f'{trial_condition}__{trial_number:03d}'
+    df.loc[row_number, 'trial_type'] = trial_name
+    return df, trial_name
+
+
+# Loop through the trials of interest and transform the DataFrame for LSS
+lss_beta_maps = {cond: [] for cond in events_df['trial_type'].unique()}
+lss_design_matrices = []
+
+for i_trial in range(events_df.shape[0]):
+    lss_events_df, trial_condition = lss_transformer(events_df, i_trial)
+
+    # Compute and collect beta maps
+    lss_glm = FirstLevelModel(**glm_parameters)
+    lss_glm.fit(fmri_file, lss_events_df)
+
+    # We will save the design matrices across trials to show them later
+    lss_design_matrices.append(lss_glm.design_matrices_[0])
+
+    beta_map = lss_glm.compute_contrast(
+        trial_condition,
+        output_type='effect_size',
+    )
+
+    # Drop the trial number from the condition name to get the original name
+    condition_name = trial_condition.split('__')[0]
+    lss_beta_maps[condition_name].append(beta_map)
+
+# We can concatenate the lists of 3D maps into a single 4D beta series for
+# each condition, if we want
+lss_beta_maps = {
+    name: image.concat_imgs(maps) for name, maps in lss_beta_maps.items()
+}
+
+##############################################################################
+# Show the design matrices for the first few trials
+# `````````````````````````````````````````````````
+fig, axes = plt.subplots(ncols=3, figsize=(20, 10))
+for i_trial in range(3):
+    plotting.plot_design_matrix(
+        lss_design_matrices[i_trial],
+        ax=axes[i_trial],
+    )
+    axes[i_trial].set_title(f'Trial {i_trial + 1}')
+
+fig.show()
+
+##############################################################################
+# Compare the three modeling approaches
+# -------------------------------------
+
+fig, axes = plt.subplots(
```

Comment:
Probably less important, and maybe even a choice of yours to increase readability, but this code cell and the next one could be factorized with a for loop. Feel free to ignore this comment if you think that the code is easier to read like this.